x-pack/plugins/uptime/public/components/overview/alerts/uptime_alerts_flyout_wrapper.tsx

```diff
@@ -10,21 +10,21 @@ import { AlertAdd } from '../../../../../../plugins/triggers_actions_ui/public';
 interface Props {
   alertFlyoutVisible: boolean;
   alertTypeId?: string;
-  canChangeTrigger?: boolean;
   setAlertFlyoutVisibility: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
 export const UptimeAlertsFlyoutWrapperComponent = ({
   alertFlyoutVisible,
   alertTypeId,
-  canChangeTrigger,
   setAlertFlyoutVisibility,
 }: Props) => (
   <AlertAdd
     addFlyoutVisible={alertFlyoutVisible}
     consumer="uptime"
     setAddFlyoutVisibility={setAlertFlyoutVisibility}
     alertTypeId={alertTypeId}
-    canChangeTrigger={canChangeTrigger}
+    // if we don't have an alert type pre-specified, we need to
+    // let the user choose
+    canChangeTrigger={!alertTypeId}
```

Comment:
if alert type id is 0, this will not work :)

Reply:
I'm not sure we're on the same page on this one, maybe elaborate?

If `alertTypeId` is 0, we'd supply a value of `0` to `AddAlert`'s `alertTypeId` prop, and `!0` to `canChangeTrigger`.

So `canChangeTrigger` will be `true` (i.e., let the user decide which alert they want), and we'd give 0 as the `alertTypeId`, which corresponds to none of the registered alerts. So the behavior (I ran this with 0 plugged in as the value) will cause the flyout to show alert selection:

![image](https://user-images.githubusercontent.com/18429259/80976451-7e128880-8df1-11ea-8987-7fa8b0eab56e.png)

That's the behavior I was going for, perhaps I'm misunderstanding. My reasoning is that, if the `alertTypeId` (which is optional) is falsey, we can't tell the alert flyout which alert we want it to display, and the user will need to choose.

In practice, the way our UI is set up this should never happen.

Reply:
yeah you are right, this will work, sorry for overlook on my part.